### PR TITLE
storage: fix gc issue #1021

### PR DESCRIPTION
**Pull request by disksing**
@zhangjinpeng1987 @ngaut @siddontang

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -139,25 +139,34 @@ impl<'a> MvccTxn<'a> {
     }
 
     pub fn gc(&mut self, key: &Key, safe_point: u64) -> Result<()> {
-        let mut after_safe_point = false;
+        let mut delete_flag = false;
         let mut ts: u64 = u64::max_value();
         while let Some((commit, write)) = try!(self.reader.seek_write(key, ts)) {
-            if !after_safe_point {
+            if !delete_flag {
```
Comment (ngaut) [inline src/storage/mvcc/txn.rs]: s/delete_flag/remove_older/

File: src/storage/mvcc/txn.rs [hunk 2]
```diff
@@ -296,21 +305,38 @@ mod tests {
...
         must_prewrite_delete(engine.as_ref(), b"x", b"x", 25);
         must_commit(engine.as_ref(), b"x", 25, 30);
+        must_prewrite_put(engine.as_ref(), b"x", b"x35", b"x", 35);
+        must_commit(engine.as_ref(), b"x", 35, 40);
+        must_prewrite_lock(engine.as_ref(), b"x", b"x", 45);
+        must_commit(engine.as_ref(), b"x", 45, 50);
+        must_prewrite_put(engine.as_ref(), b"x", b"x55", b"x", 55);
+        must_rollback(engine.as_ref(), b"x", 55);
+
+        // start commit command
```
Comment (ngaut) [inline src/storage/mvcc/txn.rs]: s/start/startTs/

Comment (ngaut) [inline src/storage/mvcc/txn.rs]: s/commit/commitTs/

**Comment by disksing** [comment]
PTAL @ngaut

**Comment by ngaut** [comment]
LGTM

**Pull request merged**